pgjdbc/src/main/java/org/postgresql/jdbc/PgConnection.java

```diff
@@ -1219,17 +1220,20 @@ protected void abort() {
     queryExecutor.abort();
   }
 
-  private synchronized Timer getTimer() {
+  private Timer getTimer() {
+    Timer cancelTimer = cancelTimerRef.get();
     if (cancelTimer == null) {
-      cancelTimer = Driver.getSharedTimer().getTimer();
+      cancelTimer = SharedTimer.getSharedInstance().getTimer();
+      cancelTimerRef.set(cancelTimer);
```

Comment:
This needs to be atomically set from null to new value. If it is not null when set here, the timer needs to be released to keep the count correct.